### PR TITLE
C.6: More illustrative example.

### PR DESCRIPTION
**Pull request by PiotrSliwa**
More illustrative example of bad/good scenario instead of simple example.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -3141,9 +3141,56 @@ Placing them in the same namespace as the class makes their relationship to the
...
 More precise statement of design intent, better readability, more errors caught by the compiler, more optimization opportunities.
 
-##### Example
+##### Example, bad
+
+    class Shape
+    {
+    public:
+        // ...
+        virtual int getArea() = 0;
```
Comment (mpark) [inline CppCoreGuidelines.md]: `snake_case` rather than `camelCase`

Comment (Liondance) [inline CppCoreGuidelines.md]: Thanks for saying it first, camelCase looks so ... Java to me!

Comment (PiotrSliwa) [inline CppCoreGuidelines.md]: Changed;)

**Comment by hsutter** [comment]
Actually I think we should just remove C.6 as a duplicate of Con.2. I think Con.2 is currently better and more concise, and that this belongs in the "const" section more than the "class" section.

So instead of merging this PR I have removed C.6 and folded its text into Con.2 so we don't lose anything.

Thanks for pointing out that C.6 was incomplete!

**Pull request closed without merging**